pkg/controller/cloud/node_controller.go

```diff
@@ -62,14 +61,14 @@ func NewCloudNodeController(
 	nodeInformer coreinformers.NodeInformer,
 	kubeClient clientset.Interface,
 	cloud cloudprovider.Interface,
-	nodeStatusUpdateFrequency time.Duration) *CloudNodeController {
+	nodeStatusUpdateFrequency time.Duration,
+	stopCh <-chan struct{}) *CloudNodeController {
```

Comment:
how is this different than the channel provided to Run()? I wouldn't expect to need a stop channel in the constructor, just in the Run() method. Should we move StartRecordingToSink to Run()? (same comment applies in multiple places)

Reply:
Sure, I would move them into Run.

Reply:
@liggitt If we move `StartRecordingToSink` to `Run`, we need to add a new field `eventBroadcaster` to Controller type, since it is required to both initialize recorder and call `StartRecordingToSink`. Is it ok?

Reply:
yeah, that's fine